torch/csrc/distributed/c10d/ProcessGroupNCCL.hpp

```diff
@@ -769,6 +771,11 @@ class TORCH_API ProcessGroupNCCL : public Backend {
   // is set to true.
   std::atomic<bool> enableTiming_;
 
+  // timer for watch dog thread;
+  timer_t timerId_;
```

Comment:
If timerId_ is put here, then it would be a global variable and shared by all ProcessGroups. Would it work in multi-PG case?